requester/requester.go

```diff
@@ -135,8 +135,8 @@ func (b *Work) Run() {
 	}()
 
 	b.runWorkers()
-	stopCh <- struct{}{}
 	if b.Output == "" {
+		stopCh <- struct{}{}
```

Comment:
We also shouldn't initialize progress reporting if b.Output != "". I will try to rewrite the progress handling if you don't want to deal with the current mess.